input/witnessgen.go

```diff
@@ -270,6 +378,7 @@ func (wt StandardWitnessType) WitnessGenerator(signer Signer,
 	return func(tx *wire.MsgTx, hc *txscript.TxSigHashes,
 		inputIndex int) (*Script, error) {
 
+		// TODO(roasbeef): copy the desc?
```

Comment:
should we worry about `desc.PrevOutputFetcher` here?

Reply:
`PrevOutputFetcher` is never actually written to disk, the caller (typically the sweeper) always populate it, as it needs to do so in order to make the sighashes, and you can't make the sighashes untli the entire sweep transaction is known (for sig hash all).